lib/svc/method/net-virtual

```diff
@@ -0,0 +1,563 @@
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T. All rights reserved.
+# Copyright (c) 1999, 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2017 Joyent, Inc.
+# Copyright 2017 Erigones, s. r. o.
+#
+
+#
+# This script is responsible for creating VNICs (dladm create-vnic) based on
+# configuration in /usbkey/config or provided via boot parameters. It can also
+# create (replace) the overlay_rules.json file. Overlays are created only
+# if a VNIC is configured to be over an overlay. Etherstubs are created in
+# network/physical.
+#
+# It also supports creating of new VNICs added to /usbkey/config on a running
+# system via the `refresh` parameter. However, the refresh functionality is
+# currently limited to configuration of VNICs that do not exist in the OS yet.
+#
+
+#### EXAMPLE /usbkey/config entries:
+##
+#### Create etherstubs (performed in network/physical)
+#### (comma-separated list)
+##
+## etherstub=foo0,bar1,test30
+##
+##
+#### Create /var/run/smartdc/networking/overlay_rules.json
+##
+## overlay_rule_spam="-e vxlan -p vxlan/listen_ip=0.0.0.0,vxlan/listen_port=5000 \
+##    -s direct -p direct/dest_ip=10.20.30.40 -p direct/dest_port=4444 -p mtu=1400"
+## overlay_rule_ham_eggs="-e vxlan -p vxlan/listen_ip=0.0.0.0,vxlan/listen_port=4795 \
+##    -s files -p files/config=/var/db/ham_eggs.json -p mtu=1400"
+##
+##
+#### VNICs over normal physical interfaces or aggregation
+#### nic_tag=foobar
+##
+## foobar_nic=11:22:33:aa:bb:cc
+#### Or foobar_nic=aggr0 for VNIC/nic_tag over aggregation inteface named `aggr0`
+## foobar0_ip=192.168.11.22
+## foobar0_netmask=255.255.255.0
+## foobar0_vlan_id=...
+## foobar0_mac=...
+## foobar0_mtu=...
+##
+##
+#### VNICs on etherstubs
+#### The etherstub name must be listed in the `etherstub=` setting (see above)
+#### nic_tag=<etherstub-name>
+##
+## test30_0_ip=172.16.33.44
+## test30_0_netmask=255.255.240.0
+## test30_0_vlan_id=...
+## test30_0_mac=...
+## test30_0_mtu=...
+##
+##
+#### VNICs on overlays
+#### There must be an `overlay_rule=` setting for the overlay rule (see above)
+#### nic_tag=<overlay rule name>
+##
+## ham_eggs_0_vxlan_id=1234
+## ham_eggs_0_ip=10.55.66.77
+## ham_eggs_0_netmask=255.255.0.0
+## ham_eggs_0_vlan_id=...
+## ham_eggs_0_mac=...
+## ham_eggs_0_mtu=...
+
+#######################################
+# The /usbkey/config entries are based on the following pattern:
+#
+# <nic_tag>_<integer:vnic-suffix>_ip=
+# <nic_tag>_<integer:vnic-suffix>_netmask=
```

Comment:
The colon character (`:`) is confusing here. Please consider something like this:
`<nic_tag>_<vnic-suffix_integer>_ip=`